infrastructure/async/src/main/java/tech/pegasys/teku/infrastructure/async/SafeFuture.java

```diff
@@ -321,6 +321,10 @@ private String getMessageFromException(final Throwable error) {
     if (error == null) {
       return UNKNOWN_ERROR;
     }
+    final Throwable rootCause = Throwables.getRootCause(error);
+    if (rootCause.getMessage() == null) {
+      return error.getMessage() != null ? error.getMessage() : UNKNOWN_ERROR;
+    }
     return Throwables.getRootCause(error).getMessage();
```

Comment:
We might want to handle the possible `IllegalArgumentException` that might be thrown by `Throwables.getRootCause(error)`.

Reply:
oh that would already be possible but yeh sure may as well